### ISSUE TITLE
using alternate go version broken or just confusing by #321

### ISSUE THREAD

**Issue by jmhodges** [bazelbuild/rules_go#365]
In https://github.com/bazelbuild/rules_go/pull/321, go_respositories was changed such that a provided go_linux and go_darwin respository name was used instead of the pre-specified versions. However, this doesn't seem to work because of the `:VERSION` appended to the end of them given the definitions of those repos I already have.

Here's the error that was emitted.

```
$  bazel query //path/to/some/go/target
ERROR: /private/var/tmp/_bazel_jmhodges/ee957dabdee1b1a9e1f2fe90369cf80f/external/io_bazel_rules_go/go/private/go_repositories.bzl:245:3: //external:io_bazel_rules_go_toolchain: invalid label 'golang_linux_amd64//:VERSION' in attribute 'go_linux_version' in '_go_repository_select' rule: invalid target name 'golang_linux_amd64//:VERSION': target names may not contain '//' path separators.
ERROR: /private/var/tmp/_bazel_jmhodges/ee957dabdee1b1a9e1f2fe90369cf80f/external/io_bazel_rules_go/go/private/go_repositories.bzl:245:3: //external:io_bazel_rules_go_toolchain: invalid label 'golang_darwin_amd64//:VERSION' in attribute 'go_darwin_version' in '_go_repository_select' rule: invalid target name 'golang_darwin_amd64//:VERSION': target names may not contain '//' path separators.
```

My repositories were already defined as

```
new_http_archive(
    name = "golang_linux_amd64",
    build_file_content = "",
    sha256 = "53ab94104ee3923e228a2cb2116e5e462ad3ebaeea06ff04463479d7f12d27ca",
    strip_prefix = "go",
    url = "https://storage.googleapis.com/golang/go1.8.linux-amd64.tar.gz",
)

new_http_archive(
    name = "golang_darwin_amd64",
    build_file_content = "",
    sha256 = "6fdc9f98b76a28655a8770a1fc8197acd8ef746dd4d8a60589ce19604ba2a120",
    strip_prefix = "go",
    url = "https://storage.googleapis.com/golang/go1.8.darwin-amd64.tar.gz",
)

load("@io_bazel_rules_go//go:def.bzl", "go_repositories", "new_go_repository")

go_repositories(go_linux = "@golang_linux_amd64", go_darwin = "@golang_darwin_amd64")
```

This is one bazel version 0.4.5 out of the homebrew packaging system.

(What's unfortunate is that I only had to upgrade rules_go because buildifier was renamed to buildtools. All these API breaks are making it very difficult to use bazel for go code.)

**Comment by jayconrod** [comment]
Could you verify that the repository names passed to `go_repositories` start with `@`? Your example works for me, but if I change the last line to this, I see the same error.

```bzl
go_repositories(go_linux = "golang_linux_amd64", go_darwin = "golang_darwin_amd64")
```

I'm sorry your build was broken by multiple changes. I'm sure it's frustrating. I hope at least with `go_repositories`, the improved flexibility is worthwhile.

**Comment by jayconrod** [comment]
I've tagged and released versions 0.3.4 and 0.4.2, which should provide an easier upgrade path. They just have the fix for the buildifier rename. Sorry again for the break.

**Comment by jmhodges** [comment]
Ugh, I can't repro in another workspace. I think this is some bazel's external repo caching that caught me. Closing! Thanks!
